Firmware/Marlin_main.cpp

```diff
@@ -2812,44 +2820,47 @@ static void gcode_G80()
+        auto bedCorrectHelper = [eeprom_bed_correction_valid] (char code, uint8_t *eep_address) -> int8_t {
+            if (code_seen(code)) {
+                // Verify value is within allowed range
+                int16_t temp = code_value_short();
+                if (abs(temp) > BED_ADJUSTMENT_UM_MAX) {
+                    printf_P(PSTR("%SExcessive bed leveling correction: %i microns\n"), errormagic, temp);
+                } else {
+                    return (int8_t)temp; // Value is valid, use it
+                }
+            } else if (eeprom_bed_correction_valid) {
+                return (int8_t)eeprom_read_byte(eep_address);
+            }
+            return 0;
+        };
+        const int8_t correction[4] = {
+            bedCorrectHelper('L', (uint8_t*)EEPROM_BED_CORRECTION_LEFT),
+            bedCorrectHelper('R', (uint8_t*)EEPROM_BED_CORRECTION_RIGHT),
+            bedCorrectHelper('F', (uint8_t*)EEPROM_BED_CORRECTION_FRONT),
+            bedCorrectHelper('B', (uint8_t*)EEPROM_BED_CORRECTION_REAR),
+        };
+        for (uint8_t row = 0; row < MESH_NUM_Y_POINTS; row++) {
+            for (uint8_t col = 0; col < MESH_NUM_X_POINTS; col++) {
+                constexpr float scaler = 0.001f / (MESH_NUM_X_POINTS - 1);
+                mbl.z_values[row][col] += scaler * (
+                + correction[0] * (MESH_NUM_X_POINTS - 1 - col)
+                + correction[1] * col
+                + correction[2] * (MESH_NUM_Y_POINTS - 1 - row)
+                + correction[3] * row);
             }
-        } else if (eeprom_bed_correction_valid) {
-            return (int8_t)eeprom_read_byte(eep_address);
-        }
-        return 0;
-    };
-    const int8_t correction[4] = {
-        bedCorrectHelper('L', (uint8_t*)EEPROM_BED_CORRECTION_LEFT),
-        bedCorrectHelper('R', (uint8_t*)EEPROM_BED_CORRECTION_RIGHT),
-        bedCorrectHelper('F', (uint8_t*)EEPROM_BED_CORRECTION_FRONT),
-        bedCorrectHelper('B', (uint8_t*)EEPROM_BED_CORRECTION_REAR),
-    };
-    for (uint8_t row = 0; row < MESH_NUM_Y_POINTS; row++) {
-        for (uint8_t col = 0; col < MESH_NUM_X_POINTS; col++) {
-            constexpr float scaler = 0.001f / (MESH_NUM_X_POINTS - 1);
-            mbl.z_values[row][col] += scaler * (
-              + correction[0] * (MESH_NUM_X_POINTS - 1 - col)
-              + correction[1] * col
-              + correction[2] * (MESH_NUM_Y_POINTS - 1 - row)
-              + correction[3] * row);
         }
     }
 
     mbl.upsample_3x3(); //interpolation from 3x3 to 7x7 points using largrangian polynomials while using the same array z_values[iy][ix] for storing (just coppying measured data to new destination and interpolating between them)
 
-    uint8_t useMagnetCompensation = code_seen('M') ? code_value_uint8() : eeprom_read_byte((uint8_t*)EEPROM_MBL_MAGNET_ELIMINATION);
-    if (nMeasPoints == 7 && useMagnetCompensation) {
-        mbl_magnet_elimination();
+    { // apply magnet compensation
```

Comment:
Nitpicking but isn't this bracket redundant? In C++ my understanding its done to limit the lifetime of some variables (like RAII), but I don't see how it applies here.

Reply:
I was forced to do this because I wanted to use a goto to the "restoring" code. See the exit label at the end

Reply:
With goto you can't cross variable lifetimes when jumping, so I had to restrict the lifetime to the strictest minimum for those variables. Not sure if I could somehow avoid the goto, while still performing the exit cleanup without duplicating that code

Reply:
Alright I'll resolve these bracket comments for now, I'll take a closer look at the code.